controllers/release/release_adapter.go

```diff
@@ -213,54 +214,59 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
+	binding, err := a.getSnapshotEnvironmentBinding(environment, releasePlanAdmission)
+	if err != nil && !errors.IsNotFound(err) {
+		return results.RequeueWithError(err)
+	}
 
 	patch := client.MergeFrom(a.release.DeepCopy())
-	a.release.Status.SnapshotEnvironmentBinding = fmt.Sprintf("%s%c%s", binding.Namespace, types.Separator, binding.Name)
+
+	if binding == nil {
+		binding, err := a.createSnapshotEnvironmentBinding(releasePlanAdmission, environment)
+		if err != nil {
+			return results.RequeueWithError(err)
+		}
+
+		a.logger.Info("Created/updated SnapshotEnvironmentBinding",
+			"SnapshotEnvironmentBinding.Name", binding.Name, "SnapshotEnvironmentBinding.Namespace", binding.Namespace)
+
+		a.release.Status.SnapshotEnvironmentBinding = fmt.Sprintf("%s%c%s", binding.Namespace, types.Separator, binding.Name)
+	}
 
 	return results.RequeueOnErrorOrContinue(a.client.Status().Patch(a.context, a.release, patch))
 }
 
-// createOrUpdateSnapshotEnvironmentBinding creates or updates a SnapshotEnvironmentBinding for the Release being
-// processed.
-func (a *Adapter) createOrUpdateSnapshotEnvironmentBinding(releasePlanAdmission *v1alpha1.ReleasePlanAdmission) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
-	application, components, snapshot, environment, err := a.getSnapshotEnvironmentResources(releasePlanAdmission)
-	if err != nil {
-		return nil, err
+// EnsureSnapshotEnvironmentBindingIsTracked is an operation that will ensure that the SnapshotEnvironmentBinding
+// Deployment status is tracked in the Release being processed.
+func (a *Adapter) EnsureSnapshotEnvironmentBindingIsTracked() (results.OperationResult, error) {
+	if !a.release.HasSucceeded() {
+		return results.ContinueProcessing()
 	}
 
-	// The binding information needs to be updated no matter if it already exists or not
-	binding := gitops.NewSnapshotEnvironmentBinding(components, snapshot, environment)
-
 	// Search for an existing binding
-	existingBinding, err := a.getSnapshotEnvironmentBinding(environment, releasePlanAdmission)
-	if err != nil && !errors.IsNotFound(err) {
-		return nil, err
+	binding, err := a.getSnapshotEnvironmentBindingFromReleaseStatus()
+	if err != nil {
+		return results.RequeueWithError(err)
 	}
 
-	if existingBinding == nil {
-		err = ctrl.SetControllerReference(application, binding, a.client.Scheme())
-		if err != nil {
-			return nil, err
-		}
-		return binding, a.client.Create(a.context, binding)
-	} else {
-		// We create the binding so if the owner reference is not already present, there must be a good reason for that
-		patch := client.MergeFrom(existingBinding.DeepCopy())
-		existingBinding.Spec = binding.Spec
-		return existingBinding, a.client.Patch(a.context, existingBinding, patch)
+	patch := client.MergeFrom(a.release.DeepCopy())
+
+	if binding == nil {
+		a.release.RegisterDeploymentStatus(binding)
 	}
+
+	return results.RequeueOnErrorOrContinue(a.client.Status().Patch(a.context, a.release, patch))
```

Comment:
Like in the previous one, the status is only modified within the `if`, so the patch creation and call should be inside.